### DISCUSSION TITLE
Collision Timeout

### DISCUSSION THREAD

**Discussion by Gurgel100** [bytebeamio/rumqtt#836] [Q&A] [answered]
Hello everyone,

My question is related to the `MqttState(CollisionTimeout)` error. When can this happen?
In my scenario the server only allows for a maximum of 100 inflight publishes. It can happen that I send more than 100 publish messages (with `QoS::AtLeastOnce`) in a short time and therefore the send queue overflows. When this happens it nearly always happen to give back a `MqttState(CollisionTimeout)` error. Is my assumption correct that the broker doesn't acknowledge the old packets fast enough?.

**Comment by de-sh** [comment, marked as answer]
According to [this code](https://github.com/bytebeamio/rumqtt/blob/af55848607e1d379db17734f226023310bf14123/rumqttc/src/state.rs#L394-L395) `CollisionTimeout` happens when for more than 2 cycles of the keepalive being triggered, packet that had collided still remains present in state, i.e. not sent onto network.

**Reply by Gurgel100** [reply]
Thank you for your response. I found it myself too.
